src/ArcCommander/APIs/GitAPI.fs

```diff
@@ -1,18 +1,94 @@
-﻿namespace ArcCommander.APIs
+namespace ArcCommander.APIs
 
 open System
 open ArcCommander
 open ArgumentProcessing
+open Fake.Tools.Git
 
-/// ArcCommander Configuration API functions that get executed by the configuration focused subcommand verbs
-module GitAPI =     
+module GitAPI =
 
-    // ToDo : Implement
-    let update (arcConfiguration:ArcConfiguration) (gitArgs:Map<string,Argument>) = 
-        raise (NotImplementedException())
-        |> ignore
+    let getRepoDir (arcConfiguration:ArcConfiguration) =
+        let workdir = GeneralConfiguration.getWorkDirectory arcConfiguration
+        printfn "%s" workdir
+
+        let gitDir = Fake.Tools.Git.CommandHelper.findGitDir(workdir).FullName
+        gitDir.Substring(0,gitDir.Length-4)
+
+    let update (arcConfiguration:ArcConfiguration) (gitArgs:Map<string,Argument>) =
+
+        // get repository directory
+        let repoDir = getRepoDir(arcConfiguration)
+
+        // track all untracked files
+        printfn "-----------------------------"
+        let rec getAllFiles(cDir:string) =
+            let mutable l = []
+
+            let dirs = System.IO.Directory.GetDirectories cDir |> Array.filter (fun x -> not (x.Contains ".git") ) |> List.ofSeq
+
+            l <- List.concat (dirs |> List.map (fun x -> getAllFiles x ))
+
+            let files = System.IO.Directory.GetFiles cDir |> List.ofSeq
+            l <- l @ files
+
+            l
+
+        let allFiles = getAllFiles(repoDir)
+
+        let allFilesPlusSizes = allFiles |> List.map( fun x -> x, System.IO.FileInfo(x).Length )
+
+        let trackWithAdd (file:string) =
+            if Fake.Tools.Git.CommandHelper.directRunGitCommand repoDir ("add "+file)
+            then printfn "git add %s" file
+            else printfn "[ERROR] git add %s" file
+
+        let trackWithLFS (file:string) =
+            if Fake.Tools.Git.CommandHelper.directRunGitCommand repoDir ("lfs track "+file)
+            then printfn "git lfs track %s" file; trackWithAdd (repoDir+".gitattributes")
+            else printfn "[ERROR] git lfs track %s" file
+
+        allFilesPlusSizes |> List.iter(
+            fun pair ->
+                let (file,size) = pair
+
+                if size>150000000L
```

Comment:
should be a config option

Reply:
Agreed. @JonasLukasczyk would you add this? 

You could just add an according function [here](
https://github.com/nfdi4plants/arcCommander/blob/b92dacb949dc69a3b9fa0b38beaa34f714a16291/src/ArcCommander/ArcConfiguration.fs#L85)

And then set the default value with the according key in the [config file](https://github.com/nfdi4plants/arcCommander/blob/developer/src/ArcCommander/config)

Alternatively I could do this after merging your PR